app/controllers/users_controller.rb

```diff
@@ -996,11 +996,15 @@ def destroy
   end
 
   def ignore
-    raise Discourse::NotFound unless SiteSetting.ignore_user_enabled
+    unless SiteSetting.ignore_user_enabled
+      raise Discourse::NotFound
+    end
+
+    unless current_user.id != params[:ignored_user_id] && User.where(id: params[:ignored_user_id], admin: false, moderator: false).exists?
```

Comment:
The double negative here is somewhat confusing to read. Also, this logic is probably best put into the `Guardian` class as a `can_ignore_user(user)` method. Then in here you could say `ensure_can_ignore_user!(user)`

Reply:
Sure, I can look into that 👍

Reply:
Just to be clear, the code that I can relocate in guardian is:
```
User.where(id: params[:ignored_user_id], admin: false, moderator: false).exists?
```

?

Reply:
You can also move the current user check too! A guardian knows which user it's checking for.